File: storage/rdb/src/main/java/project/dailyge/entity/country/Country.java
package project.dailyge.entity.country;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import project.dailyge.entity.BaseEntity;

import java.util.Objects;

import static lombok.AccessLevel.PROTECTED;

@Getter
@NoArgsConstructor(access = PROTECTED)
@Entity(name = "countries")
public class Country extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "kr_name")
    private String krName;

    @Column(name = "en_name")
    private String enName;

    @Column(name = "code")
    private String code;

    @Column(name = "alpha2")
    private String alpha2;

    @Column(name = "alpha3")
    private String alpha3;

    public Country(
        final Long id,
        final String krName,
        final String enName,
        final String code,
        final String alpha2,
        final String alpha3
    ) {
        this.id = id;
        this.krName = krName;
        this.enName = enName;
        this.code = code;
        this.alpha2 = alpha2;
        this.alpha3 = alpha3;
    }

    @Override
    public boolean equals(final Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null || getClass() != obj.getClass()) {
            return false;
        }
        final Country country = (Country) obj;
        return Objects.equals(code, country.code);
    }

    @Override
    public int hashCode() {
        return Objects.hash(code);
    }
}

File: storage/rdb/src/main/kotlin/project/dailyge/entity/country/Country.kt
package project.dailyge.entity.country

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import project.dailyge.entity.BaseEntity

@Entity(name = "countries")
class Country(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(name = "kr_name")
    val krName: String,

    @Column(name = "en_name")
    val enName: String,

    @Column(name = "code")
    val code: String,

    @Column(name = "alpha2")
    val alpha2: String,

    @Column(name = "alpha3")
    val alpha3: String,
) : BaseEntity() {

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is Country) {
            return false
        }
        return code == other.code
    }

    override fun hashCode(): Int = code.hashCode()
}
